tensorpack/dataflow/imgaug/noise.py

```diff
@@ -33,3 +35,17 @@ def _augment(self, img, noise):
         if self.clip:
             ret = np.clip(ret, 0, 255)
         return ret
+
+class SaltPepperNoise(ImageAugmentor):
+    def __init__(self, amount=0.05, s_vs_p=0.5):
+        super(SaltPepperNoise, self).__init__()
+        self._init(locals())
+
+    def _get_augment_params(self, img):
+        return self.rng.uniform(low=0, high=1, size=img.shape)
+
+    def _augment(self, img, param):
+        ret = img.copy()
```

Comment:
The current contract is that augmentors are allowed to modify the image in-place to be efficient. In that case the lowest-level dataflow should avoid its data being damaged by augmentors (by yielding a copy, see `dataset/cifar.py`).